vector/src/main/java/im/vector/app/SpaceStateHandlerImpl.kt

```diff
@@ -127,7 +139,15 @@ class SpaceStateHandlerImpl @Inject constructor(
                 }.launchIn(session.coroutineScope)
     }
 
-    override fun getSpaceBackstack() = spaceBackstack
+    override fun popSpaceBackstack(): String? {
+        vectorPreferences.getSpaceBackstack().toMutableList().apply {
+            val poppedSpaceId = removeLast()
+            vectorPreferences.setSpaceBackstack(this)
+            return poppedSpaceId
```

Comment:
nit - having the `return` within the `apply` feels a little odd to me, I would expect `apply` usages to augment the receiver but here we have a side effect and exit point

this might be a case where avoiding the scope could be simpler?

```kotlin
val backstack = vectorPreferences.getSpaceBackstack().toMutableList()
val poppedSpaceId = backstack.removeLast()
vectorPreferences.setSpaceBackstack(backstack)
return poppedSpaceId
```

or with the side effect declared with an `also`

```kotlin
val backstack = vectorPreferences.getSpaceBackstack().toMutableList()
return backstack.removeLast().also { vectorPreferences.setSpaceBackstack(backstack) }
```

super minor opinion for me, will leave up to you!